pinot-core/src/test/java/org/apache/pinot/queries/InnerSegmentSelectionSingleValueQueriesTest.java

```diff
@@ -80,6 +81,49 @@ public void testSelectLimitZero() {
     Assert.assertTrue(resultsBlock.getSelectionResult().isEmpty());
   }
 
+  @Test
+  public void testSelectionOrderByAgoFunction() {
+    String query = "SELECT daysSinceEpoch FROM testTable WHERE "
+        + "dateTimeConvert(daysSinceEpoch, '1:DAYS:EPOCH', '1:MILLISECONDS:EPOCH', '1:MILLISECONDS') > ago('P1D') "
+        + "ORDER BY daysSinceEpoch LIMIT 10";
+    SelectionOrderByOperator selectionOrderByOperator = getOperatorForSqlQuery(query);
```

Comment:
I think all the code between line 89 to 105 and and 109 to 124 can be extracted into a private helper function